### PR TITLE
integration specs for views and 

### PR DESCRIPTION
**Pull request by naji-97**
### for N+1 problem I have already solved it in the previous branch 

Use Capybara to write integration tests for each view in this project.

**for the User index page:**

- [ ] I can see the username of all other users.
- [ ] I can see the profile picture for each user.
- [ ] I can see the number of posts each user has written.
- [ ] When I click on a user, I am redirected to that user's show page.

**for user show page:**

- [ ] I can see the user's profile picture.
- [ ] I can see the user's username.
- [ ] I can see the number of posts the user has written.
- [ ] I can see the user's bio.
- [ ] I can see the user's first 3 posts.
- [ ] I can see a button that lets me view all of a user's posts.
- [ ] When I click a user's post, it redirects me to that post's show page.
- [ ] When I click to see all posts, it redirects me to the user's post's index page.

**User post index page:**

- [ ] I can see the user's profile picture.
- [ ] I can see the user's username.
- [ ] I can see the number of posts the user has written.
- [ ] I can see a post's title.
- [ ] I can see some of the post's body.
- [ ] I can see the first comments on a post.
- [ ] I can see how many comments a post has.
- [ ] I can see how many likes a post has.
- [ ] I can see a section for pagination if there are more posts than fit on the view.
- [ ] When I click on a post, it redirects me to that post's show page.

**Post_show page:**

- [ ] I can see the post's title.
- [ ] I can see who wrote the post.
- [ ] I can see how many comments it has.
- [ ] I can see how many likes it has.
- [ ] I can see the post body.
- [ ] I can see the username of each commentor.
- [ ] I can see the comment each commentor left.

### CODE REVIEW

File: spec/features/user_show_spec.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
+require 'rails_helper'
+
+RSpec.describe 'User Show Page', type: :feature do
+  before(:each) do
+    @user = User.create!(name: 'Example User', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', posts_counter: 3,
+                         bio: 'I am a user')
```
Comment (joel-kalema) [inline spec/features/user_show_spec.rb:6]: - [ ] Good job so far team 🥇, however, Remember to add these three tests, as we can see to the [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/intergration_spec_and_n_1_project.md#:~:text=I%20can%20see%20the%20user%27s%20first,me%20to%20that%20post%27s%20show%20page.). Kindly add them 👍 

````md
- I can see the user's first 3 posts.
- I can see a button that lets me view all of a user's posts.
- When I click a user's post, it redirects me to that post's show page.
````

**Review by joel-kalema — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi Team
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

0bd223a2b156e0eaf61a713e84c3d3b7



## To highlight :champagne: 


- Your design looks good :100: :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow GitHub flow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me @joel-kalema  when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/features/post_index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    end
+    it 'displays the post title' do
+      expect(page).to have_content(@post.title)
+    end
+    it 'displays the post comments counter' do
+      expect(page).to have_content(@post.comments_counter)
+    end
+    it 'displays the post comments' do
+      expect(page).to have_content(@post.comments_counter)
+    end
```
Comment (joel-kalema) [inline spec/features/post_index_spec.rb]: - [ ] Make sure you are not repeating the test 👍

**Review by Kamba56 — APPROVED** [review]

**Comment by naji-97** [comment]
Hi, ***@***.*** *


*You have approved my PR in the github but I can't see that in my
dashboard, Kindly make sure that you marked my PR approved in my
Dashboard*[image:
image.png]

I refreshed the dashboard page many times but still marked as pending

**Review by Whoistolu — COMMENTED** [review]
### Hi @naji-97 

Great job resolving the recommendations by the previous reviewer 👍🏽 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**